libdirections-offboard/src/test/java/com/mapbox/navigation/route/offboard/MapboxOffboardRouterTest.kt

```diff
@@ -0,0 +1,150 @@
+package com.mapbox.navigation.route.offboard
+
+import com.mapbox.api.directions.v5.models.DirectionsResponse
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.base.route.Router
+import com.mapbox.navigation.route.offboard.base.BaseTest
+import com.mapbox.navigation.route.offboard.extension.mapToRoute
+import com.mapbox.navigation.route.offboard.router.NavigationRoute
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.slot
+import io.mockk.verify
+import org.junit.Assert.assertNotNull
+import org.junit.Before
+import org.junit.Test
+import retrofit2.Callback
+import retrofit2.Response
+
+class MapboxOffboardRouterTest : BaseTest() {
+
+    companion object {
+        private const val ACCESS_TOKEN = "pk.token"
+    }
+
+    private val navigationRoute = mockk<NavigationRoute>(relaxed = true)
+    private val navigationRouteBuilder = mockk<NavigationRoute.Builder>(relaxed = true)
+    private lateinit var offboardRouter: MapboxOffboardRouter
+    private lateinit var callback: Callback<DirectionsResponse>
+    private val origin = Point.fromLngLat(1.0, 2.0)
+    private val waypoints = listOf<Point>()
+    private val destination = Point.fromLngLat(1.0, 5.0)
+
+    @Before
+    fun setUp() {
```

Comment:
👀 [_Working Effectively with Unit Tests_](https://leanpub.com/wewut/read) 
> when writing tests you should prefer DAMP (Descriptive And Maintainable Procedures) 

That includes a bunch of _procedures_ as [Inline Setup](https://leanpub.com/wewut/read#leanpub-auto-inline-setup) 😅

**1. setUp() method**
- When you remove Setup you may limit the number of times you need to check variables when you first encounter a test. So then you no longer need to worry about irrelevant instances from other tests
- It's a waste of time because you normally need to understand all the Setup each time you have to deal with the test
- Setup advocates argue because if you don't use it you normally create duplication. You can solve it using private methods or builders

Normally is good to structure the tests following the triple A 👀 https://speakerdeck.com/guardiola31337/elegant-unit-testing-mobilization-2016?slide=40 http://wiki.c2.com/?ArrangeActAssert
If the `A`rrange part gets too long we should create factory or builder methods. Problem with Setup is that hides the _Arrange_ part completely.

As said, I prefer this way because using DAMP you increase maintainability and readability of the tests. But that's my opinion and we can discuss further 😉 

On a side note, I highly recommend you reading that book (if you haven't already). It's 🔝 💯